src/fireedge/src/server/routes/api/zendesk/utils.js

```diff
@@ -0,0 +1,45 @@
+/* ------------------------------------------------------------------------- *
+ * Copyright 2002-2023, OpenNebula Project, OpenNebula Systems               *
+ *                                                                           *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may   *
+ * not use this file except in compliance with the License. You may obtain   *
+ * a copy of the License at                                                  *
+ *                                                                           *
+ * http://www.apache.org/licenses/LICENSE-2.0                                *
+ *                                                                           *
+ * Unless required by applicable law or agreed to in writing, software       *
+ * distributed under the License is distributed on an "AS IS" BASIS,         *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  *
+ * See the License for the specific language governing permissions and       *
+ * limitations under the License.                                            *
+ * ------------------------------------------------------------------------- */
+const { global } = require('window-or-global')
+
+/**
+ * Get data fireedge session.
+ *
+ * @param {string} username - username
+ * @param {string} token - pass
+ * @returns {object} user session
+ */
+const getSession = (username = '', token = '') => {
+  if (
+    username &&
+    token &&
+    global &&
+    global.users &&
+    username &&
+    global.users[username] &&
+    global.users[username].tokens
+  ) {
+    return global.users[username].tokens.find(
+      (curr = {}, index = 0) => curr.token === token
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Avoid using `==` and `===` operators with sensitive data fields like passwords, api keys, tokens etc. as they are vulnerable to timing attack. Consider using `crypto.timingSafeEqual` or `secure-compare` for time-safe string comparison.

[Learn more](https://cwe.mitre.org/data/definitions/208.html)